### PR TITLE
Reference pin search added

### PR DESCRIPTION
**Pull request by svandenb-dev**
Added reference pin search with several criterias:
- search radius
- maximum nearest pins found
- extend search on all components

### CODE REVIEW

**Review by ring630 — COMMENTED** [review]
@svandenb-dev can you also add below feature?

edbapp.components.instances["J5"].pins["19"].get_reference_pin()

File: pyaedt/edb_core/padstack.py [hunk 1]
```diff
@@ -1327,3 +1328,62 @@ def get_padstack_instance_by_net_name(self, net_name):
...
+            Default value is zero.
+        component_only : bool
+            When ``True`` limits search on component padstack instances only. When ``False`` search will be extended
+            to the entire layout.
+            Default value is ``True``.
+
+        Returns
+        -------
+        list
+            List of :class:`pyaedt.edb_core.edb_data.padstacks_data.EDBPadstackInstance`.
```
Comment (PipKat) [inline pyaedt/edb_core/padstack.py:1358]:
```suggestion
        """Search for reference pins with several criteria.
        Parameters
        ----------
        positive_pin : EDBPadstackInstance
            Pin used for evaluating the distance on the reference pins found.
        reference_net : str, optional
            Reference net. The default is ``"gnd"``.
        search_radius : float, optional
            Search radius for finding padstack instances. The default is ``5e-3``.
        max_limit : int, optional
            Maximum limit for the padstack instances found. The default is ``0``, in which
            case no limit is applied. The maximum limit value occurs on the nearest
            reference pins from the positive one that is found.
        component_only : bool, optional
            Whether to limit the search to component padstack instances only. The
            default is ``True``. When ``False``, the search is extended to the entire layout.

        Returns
        -------
        list
            List of :class:`pyaedt.edb_core.edb_data.padstacks_data.EDBPadstackInstance`.
```

File: pyaedt/edb_core/padstack.py [hunk 2]
```diff
@@ -1333,7 +1333,7 @@ def get_padstack_instance_by_net_name(self, net_name):
     def get_reference_pins(
         self, positive_pin, reference_net="gnd", search_radius=5e-3, max_limit=0, component_only=True
     ):
-        """
+        """Search for reference pins with several criteria.
```
Comment (PipKat) [inline pyaedt/edb_core/padstack.py:1336]:
```suggestion
        """Search for reference pins using given criteria.
```

File: pyaedt/edb_core/padstack.py [hunk 3]
```diff
@@ -1327,3 +1328,60 @@ def get_padstack_instance_by_net_name(self, net_name):
...
+            if not references_pins:
+                return pinlist
+        pinlist = [
+            p
+            for p in references_pins
+            if GeometryOperators.points_distance(positive_pin.position, p.position) <= search_radius
+        ]
+        if max_limit and len(pinlist) > max_limit:
+            pin_dict = {GeometryOperators.points_distance(positive_pin.position, p.position): p for p in pinlist}
+            pinlist = [pin[1] for pin in sorted(pin_dict.items())[:max_limit]]
```
Comment (ring630) [inline pyaedt/edb_core/padstack.py:1386]: if len(pinlist) <= max_limit, is pinlist ordered by distance?

File: pyaedt/edb_core/padstack.py [hunk 4]
```diff
@@ -1327,3 +1328,60 @@ def get_padstack_instance_by_net_name(self, net_name):
             if inst.net_name == net_name:
                 padstack_instances.append(inst)
         return padstack_instances
+
+    @pyaedt_function_handler()
+    def get_reference_pins(
+        self, positive_pin, reference_net="gnd", search_radius=5e-3, max_limit=0, component_only=True
+    ):
+        """Search for reference pins using given criteria.
+        Parameters
```
Comment (MaxJPRey) [inline pyaedt/edb_core/padstack.py:1337]:
```suggestion
        """Search for reference pins using given criteria.

        Parameters
```

File: pyaedt/edb_core/padstack.py (hunk 3 above)
Comment (svandenb-dev) [reply pyaedt/edb_core/padstack.py:1386]: yes

**Comment by Samuelopez-ansys** [comment]
@svandenb-dev Is this this PR finished? Or do plan to add more features? Great job!

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Pull request merged**